code/modules/food_and_drink/cakes.dm

```diff
@@ -24,62 +24,482 @@
+			schild.pixel_x = rand(-6, 6)
+			schild.pixel_y = rand(-6, 6)
+			for(var/b=1,b<=src.food_effects.len,b++)
+				if(src.food_effects[b] in schild.food_effects)
+					continue
+				schild.food_effects += src.food_effects[b]
+			schild.w_class = 1
+			schild.quality = src.quality
+			schild.name = "slice of [src.name]"
+			schild.desc = "a delicious slice of cake!"
+			schild.food_color = src.food_color
+			schild.sliced = 1
+			schild.amount = 1
+
+			schild.set_loc(get_turf(src.loc))
+		qdel(s) //cleaning up the template slice
+		if(src.clayer == 1) //qdel(src) if there was only one layer to the cake, otherwise, decrement the layer
+			qdel(src)
+		else
+			src.clayer--
+
+
+	proc/stack_cake(var/obj/item/reagent_containers/food/snacks/cake/custom/c,var/mob/user)
+		if(!(src.clayer<3))
+			return
+		if(c.clayer>=3)
+			return
+		if(c.sliced)
+			return
+
+		user.u_equip(c)
+		c.set_loc(user)
+		src.clayer++
+		src.reagents.maximum_volume += 100
+		src.amount += 10
+		c.reagents.trans_to(src,c.reagents.total_volume)
+
+		for(var/i=1,i<=c.food_effects.len,i++) //adding food effects to the src that arent already present
+			if(c.food_effects[i] in src.food_effects)
+				continue
+			src.food_effects += c.food_effects[i]
+
+		var/singlecake //logging the clayer before changing it later
+		if(c.clayer == 1)
+			singlecake = 1
+
+		for(var/i=1,i<=src.overlays.len,i++) //looking for candles and if you set a cake on top of it, it pops off!
+			if(("[src.overlay_refs[i]]" == "cake[(src.clayer)-1]-candle") || ("[src.overlay_refs[i]]" == "cake[(src.clayer)-1]-candle_lit") || ("[src.overlay_refs[i]]" == "cake[(src.clayer)-2]-candle") || ("[src.overlay_refs[i]]" == "cake[(src.clayer)-2]-candle_lit"))
+				src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+				var/obj/item/device/light/candle/can = new /obj/item/device/light/candle/small
+				can.set_loc(get_turf(src.loc))
+				user.show_text("<b>The candle pops off! Oh no!</b>","red")
+				cake_candle = 0
+				if(src.litfam)
+					src.put_out()
+				break
+
+		var/staticiterator = c.overlays.len
+		for(var/i=1,i<=staticiterator,i++) //the handling for actually adding the toppings to the cake
+			if(("[c.overlay_refs[i]]" == "base") || ("[c.overlay_refs[i]]" == "second")) //setting up base layer overlay
+				var/overlay_layer
+				if("[c.overlay_refs[i]]" == "base")
+					if(src.clayer == 2)
+						overlay_layer = "second"
+						src.amount2 = c.amount
+					else if(src.clayer == 3)
+						overlay_layer = "third"
+						src.amount3 = c.amount
+				else if("[c.overlay_refs[i]]" == "second")
+					overlay_layer = "third"
+					src.amount3 = c.amount
+					src.clayer++
+					src.reagents.maximum_volume += 100
+					src.amount += 10
+				var/image/stack = new /image('icons/obj/foodNdrink/food_dessert.dmi',"cake[src.clayer]-overlay")
+				stack.color = (c.GetOverlayImage(c.overlay_refs[i])).color
+				src.UpdateOverlays(stack,"[overlay_layer]")
```

Comment:
Linter is angry because it has no way to know if the return value of GetOverlayImage is `image`, that's why you need to shove it into a var first.
```suggestion
				var/image/ov_image = c.GetOverlayImage(c.overlay_refs[i])
				stack.color = ov_image.color
				src.UpdateOverlays(stack, overlay_layer)
```

Reply:
could also RETURN_TYPE it

Reply:
boop.